akka-actor/src/main/scala/akka/event/DeadLetterListener.scala

```diff
@@ -112,7 +113,10 @@ class DeadLetterListener extends Actor {
     val unwrapped = WrappedMessage.unwrap(d.message)
     val messageStr = unwrapped.getClass.getName
     val wrappedIn = if (d.message.isInstanceOf[WrappedMessage]) s" wrapped in [${d.message.getClass.getName}]" else ""
-    val logMessage = d match {
+
+    @tailrec def logMessage(msg: Any): String = msg match {
+      case dl: DeadLetter =>
+        logMessage(dl.message) // unwrap DeadLetter, e.g. DeadLetter(Dropped)
       case dropped: Dropped =>
```

Comment:
@johanandren For pub/sub, isn't it a rather "normal" case that there are no subscribers, and shall we therefore remove sending to dead letters?

https://github.com/akka/akka/blob/e722d104c9c10427cd001876a119ad61982cd11e/akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala#L157

Reply:
added config, default as it was: https://github.com/akka/akka/pull/32585/commits/ecadc482c73b0e0a8a8f557e7ffffd694aeac60f